### ISSUE TITLE
Building Issue [Minit 18.04 MATE / Manjaro 18.04 XFCE] 

### ISSUE THREAD

**Issue by aka-katto** [nihui/waifu2x-ncnn-vulkan#44]
I followed these instructions:

https://github.com/Tencent/ncnn/wiki/how-to-build

and files needed to build waifu2x-ncnn-vulkan were not appearing where they were needed. However, using the AUR package, the files appeared correctly. 

## Ubuntu

I did this on a fresh build on Mint 18.04, and I took screenshots of the process.

Did I do something wrong? I annotated each picture of what I was doing.

https://imgur.com/a/mPRtiOP

## Manjaro 

The same issue applied when I tried to follow the build instructions on Manjaro, where the files needed to build the cmake were not there. 

When I moved over to Manjaro [arch linux],  and I was able to 'solve' the issue by installing ncnn via the AUR package (which worked flawlessly), and then copying over the missing files and folders. 

Specifically, the waifu2x-ncnn-cmake errors were telling me the following files were missing in their respective directories, so I copied the files over from the AUR built NCNN into the tutorial-built NCNN. 

['export/lib/cmake/ncnn/ncnnConfig.cmake', 'export/lib/libcnn.a', 'export/include/ncnn']

Then I built it waifu2x-ncnn-vulkan using the tutorial-built NCNN, which seemed to fix the issue. 

## TL;DR

I tried following the ncnn build instructions on ubuntu/arch linux for waifu2x-ncnn-vulkan but ran into issues. I installed the AUR package (which worked) to find what was missing on the instructions-built version, and copied the files over, which fixed the problem.

**Comment by aka-katto** [comment]
I'm dumb as shit protobuff isn't installed

**Comment by aka-katto** [comment]
Nevermind, I ran the ncnn package through AUR again and it also gave the protobuf error. I'm sort of confused as to what's causing the issue here.

**Comment by Calinou** [comment]
On Fedora 31, I got further by installing ncnn to `/usr/local` like this:

```bash
# In the root `ncnn` folder:
cmake -Bbuild -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr/local
cmake --build build
sudo cmake --build build --target install

# In `waifu2x-ncnn-vulkan`:
cmake -Bbuild -DCMAKE_BUILD_TYPE=Release
cmake --build build
```

However, I get this error while building waifu2x-ncnn-vulkan:

```text
❯ cmake --build build -j1
[1/3] Building CXX object CMakeFiles/waifu2x-ncnn-vulkan.dir/main.cpp.o
FAILED: CMakeFiles/waifu2x-ncnn-vulkan.dir/main.cpp.o 
/usr/lib64/ccache/c++   -I. -isystem /usr/local/include/ncnn -fopenmp -O2 -DNDEBUG -fPIE   -fopenmp -MD -MT CMakeFiles/waifu2x-ncnn-vulkan.dir/main.cpp.o -MF CMakeFiles/waifu2x-ncnn-vulkan.dir/main.cpp.o.d -o CMakeFiles/waifu2x-ncnn-vulkan.dir/main.cpp.o -c ../main.cpp
In file included from ../main.cpp:63:
../waifu2x.h:35:11: error: ‘Pipeline’ in namespace ‘ncnn’ does not name a type
   35 |     ncnn::Pipeline* waifu2x_preproc;
      |           ^~~~~~~~
../waifu2x.h:36:11: error: ‘Pipeline’ in namespace ‘ncnn’ does not name a type
   36 |     ncnn::Pipeline* waifu2x_postproc;
      |           ^~~~~~~~
../main.cpp: In function ‘int main(int, char**)’:
../main.cpp:524:11: error: ‘create_gpu_instance’ is not a member of ‘ncnn’
  524 |     ncnn::create_gpu_instance();
      |           ^~~~~~~~~~~~~~~~~~~
../main.cpp:530:27: error: ‘get_gpu_count’ is not a member of ‘ncnn’; did you mean ‘get_cpu_count’?
  530 |     int gpu_count = ncnn::get_gpu_count();
      |                           ^~~~~~~~~~~~~
      |                           get_cpu_count
../main.cpp:535:15: error: ‘destroy_gpu_instance’ is not a member of ‘ncnn’
  535 |         ncnn::destroy_gpu_instance();
      |               ^~~~~~~~~~~~~~~~~~~~
../main.cpp:539:33: error: ‘get_gpu_info’ is not a member of ‘ncnn’
  539 |     int gpu_queue_count = ncnn::get_gpu_info(gpuid).compute_queue_count;
      |                                 ^~~~~~~~~~~~
../main.cpp:613:11: error: ‘destroy_gpu_instance’ is not a member of ‘ncnn’
  613 |     ncnn::destroy_gpu_instance();
      |           ^~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

**Comment by nihui** [comment]
I add a small build guide in README
https://github.com/nihui/waifu2x-ncnn-vulkan#build-from-source
